File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/MessageCell.m
//
//  MsgCell.m
//  OpenVideoCall
//
//  Created by CavanSu on 24/11/2017.
//  Copyright © 2017 Agora. All rights reserved.
//

#import "MessageCell.h"
#import "CommonExtension.h"

@interface MessageCell ()
@property (weak, nonatomic) IBOutlet UIView *colorView;
@property (weak, nonatomic) IBOutlet UILabel *messageLabel;
@end

@implementation MessageCell
- (void)setMessage:(Message *)message {
    self.colorView.backgroundColor = (message.type == MessageTypeInfo ? UIColor.AGMessageInfoColor : UIColor.AGMessageErrorColor);
    self.messageLabel.text = message.text;
}
@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/MessageCell.swift
//
//  ChatMessageCell.swift
//  OpenVideoCall
//
//  Created by GongYuhua on 16/8/15.
//  Copyright © 2016年 Agora. All rights reserved.
//

import UIKit

class MessageCell: UITableViewCell {
    
    @IBOutlet weak var colorView: UIView!
    @IBOutlet weak var messageLabel: UILabel!
    
    func set(with message: Message) {
        colorView.backgroundColor = message.type.color()
        messageLabel.text = message.text
    }
}
